src/olympia/versions/tasks.py

```diff
@@ -126,11 +211,29 @@ def extract_version_to_git(version_id, author_id=None, note=None):
     log.info('Extracting {version_id} into git backend'.format(
         version_id=version_id))
 
-    repo = AddonGitRepository.extract_and_commit_from_version(
-        version=version, author=author, note=note)
+    try:
+        repo = AddonGitRepository.extract_and_commit_from_version(
+            version=version, author=author, note=note)
+
+        log.info('Extracted {version} into {git_path}'.format(
+            version=version_id, git_path=repo.git_repository_path))
+    except BrokenRefError as err:
+        # This is needed to prevent a potential infinite loop if a broken
+        # reference is detected in `extract_addon_to_git()`, which we call
+        # later in this block.
+        if stop_on_broken_ref:
+            raise err
 
-    log.info('Extracted {version} into {git_path}'.format(
-        version=version_id, git_path=repo.git_repository_path))
+        # Reset the git hash of each version of the add-on related to the
+        # current add-on because we want to re-extract everything.
+        addon.versions.update(git_hash='')
```

Comment:
You need to use `addon.versions(manager='unfiltered_for_relations')` to also include deleted versions.

Reply:
Maybe this should be done by the `delete()` method of the `AddonGitRepository`.